### PR TITLE
convert api/role to pytest

### PR DESCRIPTION
**Pull request by peterdragun**
All test passed, except this one. Not sure what is causing this problem, but it is also failing in current automation. Test behavior wasn't changed by this PR. Needs some more investigation.
```
===================================== short test summary info =====================================
FAILED tests/foreman/api/test_role.py::TestCannedRole::test_negative_create_taxonomies_by_org_admin
======================= 1 failed, 42 passed, 7 skipped in 659.58s (0:10:59) =======================
```

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
Can one of the admins verify this patch?

File: tests/foreman/api/test_role.py [hunk 1]
```diff
@@ -186,24 +153,28 @@ def user_config(self, user):
...
+            'loc': entities.Location().create(),
+        }
+
+    @pytest.fixture
+    def filter_taxonomies(self):
+        """Create filter taxonomies"""
+        return {
+            'org': entities.Organization().create(),
+            'loc': entities.Location().create(),
+        }
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_role.py:174]: Why we creating similar object fixtures here ? can't we keep same fixture for both  `role_taxonomies` and `filter_taxonomies` with some common name as 'taxonomies' uses based on the scope of fixture ?

Comment (omkarkhatavkar) [reply tests/foreman/api/test_role.py:174]: seeing below code I can understand there are times you require both fixture simultaneously. ignore it

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Review by pondrejk — APPROVED** [review]

**Pull request merged**